### ISSUE TITLE
VS tooltip display too verbose when calling overloaded method with wrong parameter types

### ISSUE THREAD

**Issue by smoothdeveloper** [dotnet/fsharp#6578]
This is an error message improvement suggestion after seeing a picture posted in fsharp.org's slack (user Alanay).

Overload can be little tricky at times, when the overload count is too too high, it would be good to have the VS tooltip show a prettier summary.

#### Repro steps

Here is a simple case with `System.Convert.ToString` :

```fsharp
let a = System.Convert.ToString('w',4);;
```

#### Actual behavior

![image](https://user-images.githubusercontent.com/87944/56415189-d8ba6000-6241-11e9-8f20-577c790ceabc.png)

#### Expected behavior

Looking for proposals, here is a first one which would read better:

```
No overloads match for method 'ToString'.

Argument types given:
  - char
  - int
  
Available overloads:
  - Convert.ToString(value: obj, provider: IFormatProvider)
  - Convert.ToString(value: bool, provider: IFormatProvider)
  - Convert.ToString(value: bool, provider: IFormatProvider)
  - Convert.ToString(value: char, provider: IFormatProvider)
  - Convert.ToString(value: sbyte, provider: IFormatProvider)
  - Convert.ToString(value: byte, provider: IFormatProvider)
  ... exhaustive list we partially see
```

#### Workarounds

* reduce overload count
* taller screens

#### Related information 

* environment: Win10/VS2019 16.0.1
* related: #1103, #4982

**Comment by smoothdeveloper** [comment]
Relevant locations:

https://github.com/Microsoft/visualfsharp/blob/c4c1d9cf1097955a71c7cbc328a6b5c672a34bda/src/fsharp/TypeChecker.fs#L9877-L9944

https://github.com/Microsoft/visualfsharp/blob/c4c1d9cf1097955a71c7cbc328a6b5c672a34bda/src/fsharp/CompileOps.fs#L765-L770

https://github.com/Microsoft/visualfsharp/blob/c4c1d9cf1097955a71c7cbc328a6b5c672a34bda/src/fsharp/FSStrings.resx#L162-L164

**Comment by smoothdeveloper** [comment]
I'll look at this message (sample from https://github.com/Microsoft/visualfsharp/issues/2503#issuecomment-389835185):

```fsharp
open System
open System.Threading.Tasks
let makeTask () = new Task<_>(fun () -> 1)
let task = makeTask()
let rt1 = Task.Run(fun () -> task)
```

Actual:

> A unique overload for method 'Run' could not be determined based on type information prior to this program point. A type annotation may be needed. Candidates: Task.Run<'TResult>(function: Func<'TResult>) : Task<'TResult>, Task.Run<'TResult>(function: Func<Task<'TResult>>) : Task<'TResult>

It doesn't look great in VS:

![image](https://user-images.githubusercontent.com/87944/56754450-eb50ff80-6741-11e9-8a03-1cd81412577d.png)

Looking for:

```
A unique overload for method 'Run' could not be determined based on type information prior to this program point. 

A type annotation may be needed. 

Candidates: 
- Task.Run<'TResult>(function: Func<'TResult>) : Task<'TResult>
- Task.Run<'TResult>(function: Func<Task<'TResult>>) : Task<'TResult>
```

Please let me know if that makes sense.

I've made some progress in my branch (#6596) for the initial case:

![image](https://user-images.githubusercontent.com/87944/56754528-26ebc980-6742-11e9-8453-d470d0c15153.png)

There is work ahead for additional coverage of those messages as it is only lightly covered in fsharpqa tests.

More input from users with verbose overload resolution messages in advanced and simple cases would be greatly appreciated.

**Comment by smoothdeveloper** [comment]
closing this one which got merged in #6596.
